src/dbus_api/types.rs

```diff
@@ -132,6 +132,23 @@ impl ActionQueue {
         self.queue.push_back(DeferredAction::Remove(object_path))
     }
 
+    /// Push Remove actions for a path and all paths that reference it via
```

Comment:
Might as well get rid of original ```push_remove()``` and reimplement it as ```push_remove_recursive()```. There will never be any good reason to remove an object path and not to remove the object paths that it depends on that I can think of. Because if you do, it's just about certain that the next call will be a failure.